src/config/base.cpp

```diff
@@ -248,27 +504,45 @@ std::vector<std::string> ConfigBase::_merge(
             assert(((old_seqno == 0 && mine.empty()) || _config->unmerged_index() >= 1) &&
                    _config->unmerged_index() < all_hashes.size());
             set_state(ConfigState::Clean);
-            _curr_hash = all_hashes[_config->unmerged_index()];
+            _curr_hashes.clear();
+            auto& hashes = all_hashes[_config->unmerged_index()];
+            _curr_hashes.insert(hashes.begin(), hashes.end());
         }
     } else {
         // the merging affect nothing (if it had seqno would have been incremented), so don't
         // pointlessly replace the inner config object.
         assert(new_conf->unmerged_index() == 0);
```

Comment:
Just wondering whether this assert should be removed - just thinking that there are situations where a merge can occur with a config which is not going to result in a change

Eg. if you link a new device and, due to network conditions, it wasn't able to retrieve the existing configs then it would push up it's "new user" config, which would then be pulled down on your existing device, merged, and likely result in no changes right?

Reply:
Actually I just realised that at the moment we consistently merge in the current config because when you push up a new config message we don't update the `lastHash` value (so we don't unintentionally miss a config that was pushed up by another device at the same time)

Which means on the poll after pushing the current config we will always receive the current config back and try to merge it 🙃